paimon-common/src/main/java/org/apache/paimon/fs/FileIO.java

```diff
@@ -267,11 +279,20 @@ static FileIO get(Path path, CatalogContext config) throws IOException {
                                 + fallbackIO.getClass().getSimpleName()
                                 + " also cannot access this path.";
             }
-            throw new UnsupportedSchemeException(
-                    String.format(
-                            "Could not find a file io implementation for scheme '%s' in the classpath."
-                                    + "%s Hadoop FileSystem also cannot access this path '%s'.",
-                            uri.getScheme(), fallbackMsg, path));
+            UnsupportedSchemeException ex =
+                    new UnsupportedSchemeException(
+                            String.format(
+                                    "Could not find a file io implementation for scheme '%s' in the classpath."
+                                            + "%s Hadoop FileSystem also cannot access this path '%s'.",
+                                    uri.getScheme(), fallbackMsg, path));
+
+            if (localFileIOException != null) {
+                ex.addSuppressed(localFileIOException);
+            }
+            if (hadoopIOException != null) {
+                ex.addSuppressed(hadoopIOException);
+            }
```

Comment:
Save all suppressed exceptions in a `List`, so we don't have to check if each possible exception is null.